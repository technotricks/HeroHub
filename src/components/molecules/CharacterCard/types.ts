import {ItemClickCallback} from '@/types/base';
import {Character} from '@/types/character';

export type CardProps = {
  item: Character;
  onPress: ItemClickCallback;
};
