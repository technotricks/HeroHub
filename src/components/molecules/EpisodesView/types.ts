import {CharacterDetail} from '@/types/character';

export type EpisodesViewProps = {
  item?: CharacterDetail;
};
