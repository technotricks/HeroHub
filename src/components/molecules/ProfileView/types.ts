import {CharacterDetail} from '@/types/character';

export type ProfileViewProps = {
  item?: CharacterDetail;
};
