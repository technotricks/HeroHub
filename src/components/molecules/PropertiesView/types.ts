import {CharacterDetail} from '@/types/character';

export type PropertiesViewProps = {
  item?: CharacterDetail;
};
