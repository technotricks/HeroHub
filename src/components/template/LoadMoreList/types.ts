import {ItemClickCallback, LoadMoreCallback} from '@/types/base';
import {Character} from '@/types/character';

export namespace ILoadMore {
  export type Props = {
    items: Character[];
    isLoading: Boolean;
    onPress: ItemClickCallback;
    handleLoadMore: LoadMoreCallback;
  };
}
