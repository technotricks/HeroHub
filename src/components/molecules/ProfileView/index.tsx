import {
  Centered,
  ProfileImage,
  OverlayNameBottom,
  HText,
} from '@/components/atom';
import {ProfileViewProps} from './types';

const ProfileView: React.FC<ProfileViewProps> = ({item}: ProfileViewProps) => {
  const statusColor = item?.status == 'Alive' ? '#2CB9B0' : '#FE5E33';
  return (
    <Centered>
      <ProfileImage source={{uri: item?.image}} borderColor={statusColor} />
      <OverlayNameBottom bgColor={statusColor}>
        <Centered>
          <HText color="#fefefe" aligin="center" isBold fontSize="20px">
            {item?.status !== 'unknown' ? item?.status : ''}
          </HText>
        </Centered>
      </OverlayNameBottom>
    </Centered>
  );
};

export default ProfileView;
