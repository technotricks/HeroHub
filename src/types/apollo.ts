export type QueryResult<T> = {
  loading: boolean;
  data: T | undefined;
  error: Error | undefined;
};

export type Variables = {
  [key: string]: any;
};

export type QueryHookOptions = {
  fetchPolicy?:
    | 'cache-first'
    | 'cache-and-network'
    | 'network-only'
    | 'cache-only';
  variables?: Variables;
};
