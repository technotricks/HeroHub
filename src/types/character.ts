export interface Character {
  id: number;
  name: string;
  status: string;
  image: string;
  gender: string;
  type: string;
}

type NameObject = {
  name: string;
};

type Episode = {
  name: string;
  episode: string;
};

export interface CharacterDetail extends Character {
  origin: NameObject;
  location: NameObject;
  episode: Episode[];
}

type Info = {
  id: number;
  name: number;
  status: number;
};

// Create a TypeScript type for the query result
export type GetCharactersData = {
  characters: {
    results: Character[];
    info: Info;
  };
};

// Create a TypeScript type for the query result
export type GetCharacterData = {
  characters: {
    results: CharacterDetail[];
  };
};

export type FetchCharacters = (page: number) => void;
export type FetchCharacter = (id: number) => void;
