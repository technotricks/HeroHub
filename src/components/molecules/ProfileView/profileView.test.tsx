import 'react-native';
import React from 'react';

import {it, expect, describe} from '@jest/globals';

// Note: test renderer must be required after react-native.
import {create} from 'react-test-renderer';
import ProfileView from '.';
import {CharacterDetail} from '@/types/character';

describe('ProfileView component should render correctly', () => {
  const characterDetail: CharacterDetail = {
    id: 1,
    image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
    name: 'Rick Sanchez',
    gender: 'Male',
    status: 'Alive',
    origin: {
      name: 'Earth (C-137)',
    },
    location: {
      name: 'Citadel of Ricks',
    },
    episode: [
      {
        episode: 'S01E01',
        name: 'Pilot',
      },
      {
        episode: 'S01E02',
        name: 'Lawnmower Dog',
      },
    ],
    type: '',
  };

  const view = create(<ProfileView item={characterDetail} />);

  it('snapshot', () => {
    expect(view).toMatchSnapshot();
  });
});
