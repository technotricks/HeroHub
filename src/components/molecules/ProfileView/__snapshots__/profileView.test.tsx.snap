// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileView component should render correctly snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    borderColor="#2CB9B0"
    source={
      {
        "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      }
    }
    style={
      {
        "borderBottomLeftRadius": 100,
        "borderBottomRightRadius": 100,
        "borderColor": "#2CB9B0",
        "borderTopLeftRadius": 100,
        "borderTopRightRadius": 100,
        "borderWidth": 10,
        "height": 200,
        "width": 200,
      }
    }
  />
  <View
    bgColor="#2CB9B0"
    style={
      {
        "backgroundColor": "#2CB9B0",
        "borderBottomLeftRadius": 100,
        "borderBottomRightRadius": 100,
        "borderTopLeftRadius": 100,
        "borderTopRightRadius": 100,
        "bottom": 2,
        "paddingBottom": 2,
        "paddingLeft": 7,
        "paddingRight": 7,
        "paddingTop": 2,
        "position": "absolute",
        "textAlign": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        aligin="center"
        color="#fefefe"
        fontSize="20px"
        isBold={true}
        style={
          {
            "color": "#fefefe",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Alive
      </Text>
    </View>
  </View>
</View>
`;
